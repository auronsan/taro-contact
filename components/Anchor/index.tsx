import React, { ReactNode } from 'react';
import clsx from 'clsx';
import classes from './anchor.module.css';

type AnchorProps = {
  href: string;
  children: ReactNode;
  // font-size
  size?: 'inherit' | 'xs' | 'sm' | 'md' | 'lg';

  className?: string;
  style?: React.CSSProperties;

  id?: string;
};

export const Anchor: React.FC<AnchorProps> = (props: AnchorProps) => {
  const { href, children, size, className, style, id } = props;
  return (
    <a
      href={href}
      className={clsx(className, { [classes.root]: true, [classes[`fs-${size}`]]: !!size })}
      style={style}
      id={id}
      data-testid={id}
    >
      {children}
    </a>
  );
};
