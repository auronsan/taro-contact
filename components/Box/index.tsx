import { MouseEventHandler } from 'react';

export type BoxProps = {
  /** Class added to the root element, if applicable */
  className?: string;
  /** Inline style added to root component element*/
  style?: React.CSSProperties;
  component?: keyof JSX.IntrinsicElements;
  children?: React.ReactNode;

  // flexbox
  flex?: number;

  // margin
  m?: number;

  // margin-top
  mt?: number;

  // padding
  p?: number;

  // padding-x
  px?: number;

  // padding-y
  py?: number;

  // onclick
  onClick?: MouseEventHandler;

  // role
  role?: string;

  id?: string;
};

export const Box = ({
  children,
  className = '',
  component = 'div',
  flex,
  style,
  m,
  p,
  px,
  py,
  mt,
  role,
  id,
  ...rest
}: BoxProps) => {
  const Component = component;

  return (
    <Component
      {...(className ? { className } : {})}
      role={role}
      id={id}
      data-testid={id}
      style={{
        flex,
        ...style,
        ...(typeof m === 'number' ? { margin: `${m}px` } : {}),
        ...(typeof p === 'number' ? { padding: `${p}px` } : {}),
        ...(typeof px === 'number' ? { paddingLeft: `${px}px`, paddingRight: `${px}px` } : {}),
        ...(typeof py === 'number' ? { paddingTop: `${py}px`, paddingBottom: `${py}px` } : {}),
        ...(typeof mt === 'number' ? { marginTop: `${mt}px` } : {}),
      }}
      {...rest}
    >
      {children}
    </Component>
  );
};
