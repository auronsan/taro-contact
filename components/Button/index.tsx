import { DOMAttributes } from 'react';
import clsx from 'clsx';
import classes from './button.module.css';

export type ButtonProps = DOMAttributes<HTMLButtonElement> & {
  className?: string;
  children: React.ReactNode;
  // color
  c?: 'primary' | 'red' | 'yellow';

  // variant
  v?: 'outline' | 'filled';

  // flex
  flex?: number;

  style?: React.CSSProperties;
};

export const Button = ({
  children,
  className,
  c = 'primary',
  v = 'filled',
  flex,
  style,
  ...rest
}: ButtonProps) => (
  <button
    type="button"
    className={clsx(
      classes.root,
      {
        [classes[`button-${c}`]]: !!c,
        [classes[`button-variant-${v}`]]: !!v,
      },
      className
    )}
    style={{
      ...(flex ? { flex } : {}),
      ...style,
    }}
    {...rest}
  >
    {children}
  </button>
);
