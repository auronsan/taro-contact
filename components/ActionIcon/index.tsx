import { Box } from '@/components/Box';
import classes from './actionIcon.module.css';

type TActionIconProps = {
  onClick?: () => void;
  children?: React.ReactNode;
};
export const ActionIcon = (props: TActionIconProps): React.ReactElement => {
  const { onClick, children } = props;

  return (
    <Box
      className={classes.root}
      onClick={(e) => {
        e.stopPropagation();
        e.preventDefault();
        if (typeof onClick === 'function') {
          onClick();
        }
      }}
    >
      {children}
    </Box>
  );
};
