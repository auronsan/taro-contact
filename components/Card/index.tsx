import React from 'react';
import { Box } from '@/components/Box';
import classes from './card.module.css';

/**
 * Props for the Card component.
 */
type TCardProps = {
  /** The content to be rendered inside the card. */
  children: React.ReactNode;
};

/**
 * Card component.
 * @param props - The props for the component.
 * @returns The rendered card.
 */
export const Card = ({ children }: TCardProps): React.ReactElement => (
  <Box className={classes.card}>
    <Box>{children}</Box>
  </Box>
);
