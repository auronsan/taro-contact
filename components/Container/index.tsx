import clsx from 'clsx';
import type { BoxProps } from '@/components/Box';
import { Box } from '@/components/Box';
import classes from './container.module.css';

export interface ContainerProps extends BoxProps {
  /** Sets `max-width` of the container, value is not responsive â€“ it is the same for all screen sizes. Numbers are converted to rem. Ignored when `fluid` prop is set. `'md'` by default */
  size?: (string & {}) | number;

  /** Determines whether the container should take 100% of its parent width. If set, `size` prop is ignored. `false` by default. */
  fluid?: boolean;
}

export const Container = (props: ContainerProps) => {
  const { fluid, ...rest } = props;
  return (
    <Box
      {...rest}
      className={clsx({
        [classes.root]: true,
        [classes.fluid]: !!fluid,
      })}
    />
  );
};
