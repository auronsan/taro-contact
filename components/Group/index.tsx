import clsx from 'clsx';
import { Box } from '@/components/Box';
import type { BoxProps } from '@/components/Box';
import classes from './group.module.css';

type GroupProps = {
  justify?: 'center' | 'space-between' | 'end';
  wrap?: 'wrap' | 'nowrap';
  gap?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
  // padding
  p?: number;
} & BoxProps;

export const Group = ({ children, justify, wrap, gap = 'sm', p = 0, ...restProps }: GroupProps) => (
  <Box
    className={clsx(classes.root, {
      [classes[`justify-${justify}`]]: !!justify,
      [classes[`group-${wrap}`]]: !!wrap,
      [classes[`gap-${gap}`]]: !!gap,
    })}
    style={{ padding: p }}
    {...restProps}
  >
    {children}
  </Box>
);
