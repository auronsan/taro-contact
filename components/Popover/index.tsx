import React, { useEffect, useState } from 'react';
import { IconSettings } from '@tabler/icons-react';
import clsx from 'clsx';
import { ActionIcon } from '../ActionIcon';
import classes from './popover.module.css';

export const Popover = (props: { children: React.ReactNode }) => {
  const { children } = props;
  const [visible, setVisible] = useState(false);

  const handleClickOutside = (event: MouseEvent) => {
    if (visible && !(event.target as Element).closest(`.${classes.content}`)) {
      setVisible(false);
    }
  };

  useEffect(() => {
    document.addEventListener('mousedown', handleClickOutside);
    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, [visible]);

  return (
    <div className={classes.wrapper}>
      <ActionIcon
        onClick={() => {
          setVisible(!visible);
        }}
      >
        <IconSettings />
      </ActionIcon>
      <div
        className={clsx(classes.content, {
          [classes.visible]: visible,
        })}
      >
        {children}
      </div>
    </div>
  );
};
