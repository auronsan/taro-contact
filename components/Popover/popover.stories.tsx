import React from 'react';
import { Group } from '@/components/Group';
import { Popover } from './index';

export default {
  title: 'components/Popover',
  component: Popover,
};

export const Default = () => (
  <Group justify="center">
    <Popover target="Target" targetId="popover-target">
      <p>Popover content goes here</p>
    </Popover>
  </Group>
);
