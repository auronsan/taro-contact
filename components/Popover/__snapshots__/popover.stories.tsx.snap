// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Popover Default smoke-test 1`] = `
<main class="roboto-normal">
  <div style="display: flex; justify-content: center; align-items: center; height: 100vh; flex-direction: column;">
    <div class="group_root__JMRrN group_justify-center__JoiHJ group_gap-sm__VCWmB">
      <div class="popover_wrapper__jeMey">
        <div class="actionIcon_root__jjWKK"
             role="button"
             id="popover-target"
             data-testid="popover-target"
        >
          Target
        </div>
        <div class="popover_content__Av33o">
          <p>
            Popover content goes here
          </p>
        </div>
      </div>
    </div>
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;
